tests/fixtures/cloud_uploader.py

```diff
@@ -128,8 +133,12 @@ def upload_file(
             *args,
             **kwargs) -> None:
         logger.info("%s", f"Uploading {local_path} to gs://{self.bucket.name}/{remote_path}")
+        fp = os.path.join(self.local_root, local_path)
         blob = self.bucket.blob(remote_path)
-        blob.upload_from_filename(os.path.join(self.local_root, local_path))
+        blob.upload_from_filename(fp)
         if metadata_keys:
```

Comment:
```
>>> a = {}
>>> if a:
...     print("Hello")
... 
```

i would suggest `if metadata_keys is not None:`

Reply:
Uff, perhaps should check for len(metadata_keys) given my log comment.

Reply:
```
>>> len(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
>>> 
```

Reply:
Maybe overthinking, but passing in `metadata_keys={}` means that `metadata_keys is not None` evaluates as True.

Perhaps:
def upload_file(...):
    if metadata_keys is None:
        metadata_keys = dict()
    ...
    if len(metadata_keys):
        # Add size tag when other metadata is present
        sz = os.stat(fp).st_size
        metadata_keys['hca-dss-size'] = str(sz)
    blob.metadata = metadata_keys
    blob.patch()

Reply:
Yeah, that's fine.  If you want to do that, I fully support it. :)